pandas/algos.pyx

```diff
@@ -286,6 +301,80 @@ def rank_1d_int64(object in_arr, ties_method='average', ascending=True,
         return ranks
 
 
+def rank_1d_uint64(object in_arr, ties_method='average', ascending=True,
```

Comment:
same we need to create a algos_rank_helper.pxi.in and move all of the rank routines (and then add uint64). prob easiest to move, then add (in another PR), but up to you.